api/capi/src/nntrainer.cpp

```diff
@@ -888,7 +888,41 @@ ml_train_dataset_set_property_for_mode_(ml_train_dataset_h dataset,
         return status_;
       }
 
-      status_ = db->setProperty(args);
+      std::vector<std::string> properties;
+      for (unsigned int i = 0; i < args.size(); ++i) {
+        char *key_ptr = (char *)args[i];
+        std::string key = key_ptr;
+        std::string value;
+
+        /** Handle the user_data as a special case, serialize the address and
+         * pass it to the databuffer */
+        if (key == "user_data") {
+          /** This ensures that a valid user_data element is passed by the user
+           */
+          if (i + 1 >= args.size()) {
+            ml_loge("key user_data expects, next value to be a pointer");
+            status_ = ML_ERROR_INVALID_PARAMETER;
+            return status_;
+          }
+          std::ostringstream ss;
+          ss << key << '=' << args[i + 1];
+          properties.push_back(ss.str());
+
+          /** As values of i+1 is consumed, increase i by 1 */
+          i++;
+
+        } else if (key.find("user_data") == 0) {
```

Comment:
this adds a limitation for the future properties which can include the term "user_data" in them.
recommend to rather use `key.startsWith("user_data=")` after removing all white spaces from the key.

Reply:
Nice point, I have change to startsWith here 👍